syntax/grammar

```diff
@@ -0,0 +1,52 @@
+
+use_statement: identifier ['as' identifier]
+
+function_def: 'fn' NAME '(' decl_list ')' type NEWLINE function_body
+
+return_statement: 'ret' [expression]
+
+decl_list: decl | decl "," decl_list | epsilon
+
+type: i64 | i32 | i16 | i8 | i1 | f32 | f64 | str | '[' type ']' | void | NAME
```

Comment:
As well, just want to confirm here: array types are their own trait-able type. They're not just a collection of a type, they are their own individual types. Is this represented accurately here?